File: API/Models/Transaction.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using API.Enumerations;

namespace API.Models
{
    public class Transaction
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int Id { get; set; }
        public string BeneficiaryName { get; set; }
        public DateTime Date { get; set; }
        public Direction Direction { get; set; }
        public double Amount { get; set; }
        public string Description { get; set; }
        public string Currency { get; set; }

        [ForeignKey("MccCodes")]
        public int? MCC { get; set; }
        public TransactionKind Kind { get; set; }
        [ForeignKey("Category")]
        public string? CatCode { get; set; }
        public Category? Category { get; set; }
        public List<TransactionSplit> Splits { get; set; }
    }
}

File: client/src/app/core/models/transaction.ts
import { Category } from './category';
import { Kind } from './kind';

export interface Transaction {
  id: number;
  beneficiaryName?: string;
  date: Date;
  direction: string;
  amount: number;
  description: string;
  currency: string;
  mcc?: number;
  kind: Kind;
  category?: Category;
}
